### PR TITLE
Bug 1868976: jsonnet: configure SCCs

### PR DESCRIPTION
**Pull request by s-urbaniak**
* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by lilic** [comment]
/retest

doesn't seem related.

**Comment by s-urbaniak** [comment]
/cc @openshift/openshift-team-monitoring 

side note: i am in parallel working out where to place a sensible upgrade test for this in origin/e2e. I guess we cannot do an upgrade test here.

**Comment by s-urbaniak** [comment]
fwiw we already have an e2e test that at least tests Prometheus PVC: https://github.com/openshift/cluster-monitoring-operator/blob/f4b5b9b15b2691a1d6499e9587d32c260d5dbcbd/test/e2e/prometheus_test.go#L30-L88

do you think it makes sense to add more PVC tests here, i.e. for alertmanager and user-workload-prometheus?

File: jsonnet/alertmanager.jsonnet [hunk 1]
```diff
@@ -21,6 +21,16 @@ local authorizationRole = policyRule.new() +
                           ]) +
                           policyRule.withVerbs(['create']);
 
+local sccRole = policyRule.new() +
```
Comment (simonpasquier) [inline jsonnet/alertmanager.jsonnet:24]: It would make sense to have a comment IMO.

File: jsonnet/prometheus-user-workload.jsonnet [hunk 2]
```diff
@@ -54,6 +54,18 @@ local alertmanagerRole =
   ]) +
   policyRule.withVerbs(['get']);
 
+local sccRole =
```
Comment (simonpasquier) [inline jsonnet/prometheus-user-workload.jsonnet:57]: ditto

**Review by simonpasquier — COMMENTED** [review]
Agree that additional tests for Alertmanager and UWM would be good to have.

File: jsonnet/alertmanager.jsonnet (hunk 1 above)
Comment (s-urbaniak) [reply jsonnet/alertmanager.jsonnet:24]: agreed, this is a good point.

File: jsonnet/prometheus-user-workload.jsonnet (hunk 2 above)
Comment (s-urbaniak) [reply jsonnet/prometheus-user-workload.jsonnet:57]: agreed

**Comment by s-urbaniak** [comment]
@simonpasquier thanks for the suggestion! ptal if the place/wording lgty. I would submit the additional e2e tests as a follow-up if possible.

**Comment by s-urbaniak** [comment]
looking into the e2e failure, it seems to be related.
```
    --- FAIL: TestUserWorkloadMonitoring/enable_user_workload_monitoring,_assert_prometheus_rollout#01 (0.01s)
```

**Comment by s-urbaniak** [comment]
failure does seem to be unrelated, the failing test function fails at creating the configmap to turn on user workload monitoring:
```
    user_workload_monitoring_test.go:139: retrieving ConfigMap object failed: Get "https://api.ci-op-w6vf2c6m-a317e.origin-ci-int-aws.dev.rhcloud.com:6443/api/v1/namespaces/openshift-monitoring/configmaps/cluster-monitoring-config": dial tcp 54.243.39.215:6443: connect: connection refused
=== CONT  TestUserWorkloadMonitoring
    user_workload_monitoring_test.go:96: scenario "enable user workload monitoring, assert prometheus rollout" failed
...
```

**Comment by s-urbaniak** [comment]
thanos ruler is missing, thank you @simonpasquier for the OOB catch :+1:

**Comment by s-urbaniak** [comment]
the more i think about it though, the more i believe we should add those e2e tests here :thinking: we are adding quite some settings to those statefulsets here, so let me amend the tests in this very PR.

**Comment by s-urbaniak** [comment]
I just found we already have PVC tests for alertmanager :sweat_smile:

**Comment by s-urbaniak** [comment]
And we have also PVC tests for UWM Prometheus, adding a dedicated one for Thanos Ruler.

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Pull request merged**